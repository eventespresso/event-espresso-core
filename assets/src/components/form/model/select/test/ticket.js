import TicketSelect, { EditorTicketSelect } from '../ticket';
import { shallow, render } from 'enzyme';

jest.mock('../../base/model-select', () => () => <span>TicketSelect</span>);

describe('TicketSelect()', () => {
	const expectedDefaultQueryData = {
		limit: 100,
		order: 'DESC',
		orderBy: 'start_date',
		showExpired: false,
	};
	it('matches snapshot with default props', () => {
		const wrapper = shallow(<TicketSelect />);
		expect(wrapper).toMatchSnapshot();
	});
	it('matches expected default props', () => {
		const wrapper = shallow(<TicketSelect />);
		expect(wrapper.prop('queryData')).toEqual(expectedDefaultQueryData);
		expect(wrapper.prop('selectConfiguration').placeholder).toEqual(
			'Select Ticketâ€¦'
		);
		expect(wrapper.prop('modelName')).toEqual('ticket');
		expect(wrapper.prop('label')).toEqual('Select Ticket');
	});
});

describe('EditorTicketSelect()', () => {
	it('matches snapshot with default props', () => {
		const wrapper = render(<EditorTicketSelect />);
		expect(wrapper).toMatchSnapshot();
	});
});
