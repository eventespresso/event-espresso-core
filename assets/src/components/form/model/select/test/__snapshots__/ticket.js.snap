// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorTicketSelect() matches snapshot with default props 1`] = `
<div
  class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
>
  <div
    class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
  >
    <label
      class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
      for="inspector-select-ticket-control-0"
    >
      Select Ticket
    </label>
    <span>
      TicketSelect
    </span>
  </div>
</div>
`;

exports[`TicketSelect() matches snapshot with default props 1`] = `
<Component
  getQueryString={[Function]}
  label="Select Ticket"
  modelName="ticket"
  optionsEntityMap={
    Object {
      "default": Object {
        "label": "TKT_name",
        "value": "TKT_ID",
      },
    }
  }
  queryData={
    Object {
      "limit": 100,
      "order": "DESC",
      "orderBy": "start_date",
      "showExpired": false,
    }
  }
  selectConfiguration={
    Object {
      "defaultValue": undefined,
      "loadingMessage": [Function],
      "noOptionsMessage": [Function],
      "onChange": undefined,
      "placeholder": "Select Ticketâ€¦",
    }
  }
/>
`;
